### PR TITLE
[MINOR] Code Cleanup (Clients Module)

### PR DESCRIPTION
**Pull request by sjhajharia**
## What
Code Cleanup in Clients Module

## Changes
Some common changes include

- Replace the Arrays.asList() with Collections.singletonList() wherever possible
- Cleaning up some if-else blocks
- Replacing some instances of String.builder() with String operations

Thanks!

### CODE REVIEW

**Comment by chia7712** [comment]
@sjhajharia Could you please rebase code to include #16044? It fixes a lot of failed tests :)

**Comment by sjhajharia** [comment]
Thanks for pointing out @chia7712 
I rebased the PR!

**Comment by chia7712** [comment]
> I rebased the PR!

I guess other PRs will encounter same failed tests, so please rebase code for others too, thanks!

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 1]
```diff
@@ -371,6 +371,7 @@ public static boolean isEqualConstantTime(char[] first, char[] second) {
             int j = i < second.length ? i : 0;
             if (first[i] != second[j]) {
                 matches = false;
+                break;
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: This is expected behavior. see the comment `time-constant comparison that always compares all characters in first array`

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java [hunk 2]
```diff
@@ -1336,13 +1335,13 @@ public ClientResponse mockOffsetCommitResponse(String topic,
...
         OffsetCommitResponseData responseData = new OffsetCommitResponseData()
-            .setTopics(Arrays.asList(
-                new OffsetCommitResponseData.OffsetCommitResponseTopic()
-                    .setName(topic)
-                    .setPartitions(Collections.singletonList(
-                        new OffsetCommitResponseData.OffsetCommitResponsePartition()
-                            .setErrorCode(error.code())
-                            .setPartitionIndex(partition)))));
+            .setTopics(Collections.singletonList(
+                    new OffsetCommitResponseData.OffsetCommitResponseTopic()
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java]: Could you please remove those unrelated changes? That can reduce the size of PR and avoid unnecessary conflicts to other PRs

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java [hunk 3]
```diff
@@ -311,7 +311,7 @@ public void testPerformAssignmentShouldUpdateGroupSubscriptionAfterAssignmentIfN
             List<Collection<String>> capturedTopics = topicsCaptor.getAllValues();
 
             // expected the final group subscribed topics to be updated to "topic1"
-            Set<String> expectedTopicsGotCalled = new HashSet<>(Arrays.asList(topic1));
+            Set<String> expectedTopicsGotCalled = new HashSet<>(singletonList(topic1));
             assertEquals(expectedTopicsGotCalled, capturedTopics.get(0));
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java]: How about `assertEquals(Collections.singleton(topic1), capturedTopics.get(0));`?

**Review by chia7712 — COMMENTED** [review]
@sjhajharia thanks for updated PR

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java (hunk 1 above)
Comment (sjhajharia) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: Thanks for pointing that out!

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java (hunk 2 above)
Comment (sjhajharia) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java]: Done! Thanks for pointing out.

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java (hunk 3 above)
Comment (sjhajharia) [reply clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java]: Sounds good! Updated.

**Review by chia7712 — APPROVED** [review]
LGTM
